File: Arrays/valid_anagram/valid_anagram.py


def isAnagram(s: str, t: str) -> bool:
        
        freq_map = [0] * 26
        
        idx = lambda x: ord(x) - 97
        
        for i in range(len(s)):
            freq_map[idx(s[i])] += 1
            
        for j in range(len(t)):
            freq_map[idx(t[j])] -= 1
        
        if [0] * 26 == freq_map:
            return True
        return False


def stdout(s, t):
    print(f's : {s}')
    print(f't : {t}')

    print(f'is anagram : {isAnagram(s, t)}')

s = "anagram"
t = "nagaram"

stdout(s, t)

s = "car"
t = "rat"

stdout(s, t)

File: Arrays/valid_anagram/valid_anagram.js


function valid_anagram(s, t) {

    let hashmap = new Uint8Array(26);
    
    if (s.length != t.length) {
        return false;
    }
    
    function idx(x) {
        return x.charCodeAt() - 97;
    }
    
    for (let i = 0; i < s.length; i++) {
        hashmap[idx(s[i])]++;
        hashmap[idx(t[i])]--;
    }
    
    for (i = 0; i < 26; i++) {
        if (hashmap[i] != 0) {
            return false;
        }
    }
    return true;
}

let s = "anagram";
let t = "nagaram";

console.log('s :', s);
console.log('t :', t);

console.log('valid anagram: ', valid_anagram(s, t));

s = "car";
t = "rat";

console.log('s :', s);
console.log('t :', t);

console.log('valid anagram: ', valid_anagram(s, t));